srcpkgs/rpi-kernel/template

```diff
@@ -116,24 +194,30 @@ do_build() {
 	make ${makejobs} ${_cross} ARCH=${_arch} prepare
 	make ${makejobs} ${_cross} ARCH=${_arch} ${target}
 }
-do_install() {
+
+_install() {
+	local rpi_flavor="$1"
 	local hdrdest
 
 	# Run depmod after compressing modules.
-	sed -i '2iexit 0' scripts/depmod.sh
+	vsed -i '2iexit 0' scripts/depmod.sh
 
 	# Install kernel, firmware and modules
 	make ${makejobs} ARCH=${_arch} INSTALL_MOD_PATH=${DESTDIR} modules_install
 
 	# Install device tree blobs
 	make ${makejobs} ARCH=${_arch} INSTALL_DTBS_PATH=${DESTDIR}/boot dtbs_install
 
-	#move rpi3's dtb that ended up in /boot/broadcom
-	case "$XBPS_TARGET_MACHINE" in
-	aarch64*)
-		mv ${DESTDIR}/boot/broadcom/bcm2710-rpi-3-b.dtb ${DESTDIR}/boot
-		;;
+	#move dtb that ended up in /boot/broadcom
+	case "$rpi_flavor" in
+		rpi3)
+			mv ${DESTDIR}/boot/broadcom/bcm2710-rpi-3-b.dtb ${DESTDIR}/boot
+			;;
+		rpi4)
+			mv ${DESTDIR}/boot/broadcom/bcm2711-rpi-4-b.dtb ${DESTDIR}/boot
+			;;
 	esac
+	rm -rf ${DESTDIR}/boot/broadcom
```

Comment:
This block causes problems because some dtb files are missing in `rpi3-kernel` and `rpi4-kernel` (the same problem affects the current `rpi3-kernel` package). For example, on my 3b+ which tries to use NFS root with kernel IP configuration, the Ethernet adapter gets a new, random MAC address with every boot. The `smsc95xx.macaddr` kernel command-line argument makes no difference. The problem is resolved by copying `/boot/broadcom/bcm2710-rpi-3-b-plus.dtb` and `/boot/broadcom/bcm2837-rpi-3-b-plus.dtb` to `/boot`.

I think the right thing here is to copy all of the relevant family and SoC dtb files (see [this table](https://openwrt.org/toh/raspberry_pi_foundation/raspberry_pi#hardware)):

```bash
case "$rpi_flavor in
	rpi3)
		mv ${DESTDIR}/boot/broadcom/bcm{2710,2837}-*.dtb ${DESTDIR}/boot
		;;
	rpi4)
		mv ${DESTDIR}/boot/broadcom/bcm2711-*.dtb ${DESTDIR}/boot
		;;
esac
rm -rf ${DESTDIR}/broadcom
```
This should capture all of the proper DTBs for `rpi3`-class hardware. For `rpi4`, there is currently only one DTB, but it looks like there are more `bcm2711-*` DTBs for the RPi 400 and CM4 [coming in 5.10.x](https://github.com/raspberrypi/linux/tree/rpi-5.10.y/arch/arm/boot/dts), so doing a glob probably saves us some trouble later.

Reply:
good catch @ahesford , thank you